This can't be upgraded because it needs java upgrade.
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/jenkins/agent/workspace/ade-of-closure-compiler-unshaded/common/src/main/java/org/broadleafcommerce/common/resource/service/GoogleClosureJavascriptMinificationServiceImpl.java:[25,36] cannot access com.google.javascript.jscomp.AbstractCommandLineRunner
  bad class file: /home/jenkins/.m2/repository/com/google/javascript/closure-compiler-unshaded/v20230802/closure-compiler-unshaded-v20230802.jar(com/google/javascript/jscomp/AbstractCommandLineRunner.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
```

--- a/common/pom.xml
+++ b/common/pom.xml
@@ -420,5 +420,13 @@
             <groupId>com.googlecode.htmlcompressor</groupId>
             <artifactId>htmlcompressor</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.protobuf</groupId>
+            <artifactId>protobuf-java</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+        </dependency>
     </dependencies>
 </project>

--- a/common/src/main/java/org/broadleafcommerce/common/resource/service/GoogleClosureJavascriptMinificationServiceImpl.java
+++ b/common/src/main/java/org/broadleafcommerce/common/resource/service/GoogleClosureJavascriptMinificationServiceImpl.java
@@ -17,6 +17,7 @@
  */
 package org.broadleafcommerce.common.resource.service;
 
+import com.google.common.io.CharStreams;
 import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
@@ -96,7 +97,7 @@ public class GoogleClosureJavascriptMinificationServiceImpl implements Javascrip
     @Override
     public void minifyJs(String filename, Reader reader, Writer writer) throws ResourceMinificationException {
         try {
-            SourceFile input = SourceFile.fromReader(filename, reader);
+            SourceFile input = SourceFile.fromCode(filename, CharStreams.toString(reader));
             String compiled = compileJs(input, filename);
             writer.write(compiled);
         } catch (IOException e) {

--- a/pom.xml
+++ b/pom.xml
@@ -58,7 +58,7 @@
         <gpg.plugin.version>1.6</gpg.plugin.version>
         <broadleaf-presentation.version>1.3.5-GA</broadleaf-presentation.version>
         <hsqldb.database.starter.version>2.3.4-GA</hsqldb.database.starter.version>
-        <closure.compiler.version>v20210505</closure.compiler.version>
+        <closure.compiler.version>v20200830</closure.compiler.version>
         <hibernate.version>5.4.33.Final</hibernate.version>
         <hibernate-validator.version>6.1.7.Final</hibernate-validator.version>
         <javax.version>2.3.1</javax.version>

